source/grid/manifold_lib.cc

```diff
@@ -301,68 +301,146 @@ get_new_point (const std::vector<Point<spacedim> > &vertices,
 // ============================================================
 
 template <int dim, int spacedim>
-CylindricalManifold<dim,spacedim>::CylindricalManifold(const unsigned int axis,
-                                                       const double tolerance) :
-  direction (Point<spacedim>::unit_vector(axis)),
-  point_on_axis (Point<spacedim>()),
-  tolerance(tolerance)
+CylindricalManifold<dim, spacedim>::CylindricalManifold(const unsigned int axis,
+                                                        const double tolerance) :
+  ChartManifold<dim,spacedim,3>(Tensor<1,3>({0,2.*numbers::PI,0})),
+              normal_direction(compute_normal(Point<3>::unit_vector(axis))),
+              direction (Point<3>::unit_vector(axis)),
+              point_on_axis (Point<3>()),
+              tolerance(tolerance)
 {
-  Assert(spacedim > 1, ExcImpossibleInDim(1));
+  static_assert(spacedim==3,
+                "CylindricalManifold can only be used for spacedim==3!");
```

Comment:
This is the wrong place for it -- you would only ever get to see the error message if you tried to change the .inst.in file to also compile the class for anything other than `spacedim==3`. I think you want to move this into the class declaration itself, so that a user gets to see the error message *while trying to use the class* for invalid template arguments.